src/pause/140C70.c

```diff
@@ -1,17 +1,287 @@
+    D_80270708 = 0.0f;
+    D_80270704 = temp_f2_2;
+    D_80270708 = temp_f0_2;
+    if (temp_f6 < temp_f2_2) {
+        D_80270704 = temp_f6;
+    }
+    if (temp_f6 < temp_f0_2) {
+        D_80270708 = temp_f6;
+    }
+    if (D_80270704 <= -86.0f) {
+        D_80270704 = -85.0f;
+    }
+    if (D_80270708 <= -210.0f) {
+        D_80270708 = -209.0f;
+    }
+    *arg0 = 1;
+}
+#else
 INCLUDE_ASM(s32, "pause/140C70", pause_map_init);
+#endif
+
+void pause_map_handle_input(void) {
+    f64 temp_f4_2;
+    s32 temp_f12;
+    pause_map_spaces* map_spaces;
+    GameStatus* gameStatus;
+    f32 phi_f8;
+    f32 phi_f6;
+    f32 mapCursorYPosTemp = mapCursorYPos;
+    f32 mapCursorXPosTemp = mapCursorXPos;
+    
+
+    gameStatus = gGameStatusPtr;
+    phi_f8 = (f32) gameStatus->stickX * 0.05f;
+    phi_f6 = (f32) -(s32) gameStatus->stickY * 0.05f;
+    if ((phi_f8 == 0.0f) && (phi_f6 == 0.0f)) {
+        if (mapCursorCurrentOption != -1) {
+            map_spaces = (pause_map_spaces*)&D_802502B8 + mapCursorCurrentOption;
+            phi_f8 = map_spaces->xPos - mapCursorXPosTemp;
+            phi_f8 = phi_f8 * D_80270090;
+            phi_f6 = map_spaces->yPos - mapCursorYPosTemp;
+            phi_f6 = phi_f6 * D_80270090;
+        }
+    }
+    mapCursorXPos += phi_f8;
+    mapCursorYPos += phi_f6;
+    if (mapCursorXPos < 16.0f) {
+        mapCursorXPos = 16.0f;
+    }
+    if (mapCursorYPos < 8.0f) {
+        mapCursorYPos = 8.0f;
+    }
+    if (mapCursorXPos >= 316.0f) {
+        mapCursorXPos = 315.0f;
+    }
+    if (mapCursorYPos >= 308.0f) {
+        mapCursorYPos = 307.0f;
+    }
+
+    temp_f4_2 = (s32) (mapCursorXPos + D_80270704 - D_80270098);
+    temp_f12 = mapCursorYPos + D_80270708 - D_802700A0;
+    if (D_802700A8 <= temp_f4_2) {
+        D_80270704 = D_80270704 - (temp_f4_2 - D_802700A8);
+    }
+
+    if (temp_f4_2 <= D_802700B0) {
+        D_80270704 = D_80270704 - (temp_f4_2 + D_802700B8);
+    }
+
+    temp_f12 = temp_f12;
+    if (temp_f12 >= 15.0) {
+        D_80270708 = D_80270708 - (temp_f12 - 15.0);
+    }
+
+    if (temp_f12 <= -15.0) {
+        D_80270708 =  D_80270708 - (temp_f12 + 15.0);
+    }
+
+    if (D_80270704 > 0.0f) {
+        D_80270704 = 0.0f;
+    }
+
+    if (D_80270708 > 0.0f) {
+        D_80270708 = 0.0f;
+    }
+
+    if (D_80270704 < -86.0f) {
+        D_80270704 = -86.0f;
+    }
+
+    if (D_80270708 < -210.0f) {
+        D_80270708 = -210.0f;
+    }
+
+    if ((gPauseMenuPressedButtons & B_BUTTON) != 0) {
+        sfx_play_sound(0xCA);
+        gPauseMenuCurrentTab = 0;
+        return;
+    }
+
+    gPauseMenuCurrentDescIconScript = 0;
+    if (mapCursorCurrentOption == -1) {
+        gPauseMenuCurrentDescString = 0;
+        return;
+    }
+
+    gPauseMenuCurrentDescString = (mapCursorCurrentOption * 3) + 0x1D00F3;
+    if (evt_get_variable(0, EVT_STORY_PROGRESS) >= (s32)*(D_802502B8 + (mapCursorCurrentOption * ARRAY_COUNT(D_802502B8) + 3))) {
```

Comment:
D_802502B8[mapCursorCurrentOption].unk_3

Reply:
This was marked as resolved, but  I would try after changing the type of D_802502B8 to be only an array, not an array of pointers.